include/armadillo_bits/diskio_meat.hpp

```diff
@@ -1305,7 +1316,8 @@ diskio::save_hdf5_binary(const Mat<eT>& x, const hdf5_name& spec)
     const std::string dataset_name = (full_name.empty() == false) ? full_name : std::string("dataset");
 
     hid_t dataset = arma_H5Dcreate(groups.size() == 0 ? file : groups[groups.size() - 1], dataset_name.c_str(), datatype, dataspace, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
-    
+    if (dataset < 0) { arma_debug_warn("Mat::save(): dataset ", dataset_name.c_str(), std::string(" already exists in ") + spec.filename.c_str());} //can never happen if file is overwritten, so no need to use tmp_name.c_str()
```

Comment:
Should this be `arma_check()` instead of `arma_debug_warn()`?  I think this should cause the program to fail in both debugging and non-debugging mode.